src/Adapter/DbSelect.php

```diff
@@ -122,7 +122,9 @@ public function count()
         $result    = $statement->execute();
         $row       = $result->current();
 
-        $this->rowCount = (int) $row[self::ROW_COUNT_COLUMN_NAME];
+        $this->rowCount = isset($row[self::ROW_COUNT_COLUMN_NAME])
+            ? (int) $row[self::ROW_COUNT_COLUMN_NAME]
+            : (int) $row[strtolower(self::ROW_COUNT_COLUMN_NAME)];
```

Comment:
What happens when the lowercase version is not present? This is likely to raise a warning.

I'd argue this should be done as follows:

```php
$this->rowCount = $this->locateRowCount($row);
```

with the following helper method:

```php
/* Not sure if `array` is correct here, or if this could be an `ArrayAccess` instance */
private function locateRowCount(array $row): int
{
    if (array_key_exists(self::ROW_COUNT_COLUMN_NAME, $row)) {
        return (int) $row[self::ROW_COUNT_COLUMN_NAME];
    }

    $lowerCaseColumnName = strtolower(self::ROW_COUNT_COLUMN_NAME);
    if (array_key_exists($lowerCaseColumnName, $row)) {
        return (int) $row[$lowerCaseColumnName];
    }

    throw new LogicException('Unable to determine row count; missing row count column in result');
}
```

This will satisfy #10, while also making the code less brittle.

Reply:
@weierophinney implemented